### PR TITLE
Fix HTTP cache assertion failure for small disks

### PR DESCRIPTION
**Pull request by GregoryTravis**
The HTTP cache was attempting to make room for the largest allowed file. In cases where that was smaller than the largest allowed total cache size, there was an assertion failure.

Also added the actual size to the `Response_Too_Large` error.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.
- [x] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso [hunk 1]
```diff
@@ -560,24 +560,25 @@ type Out_Of_Range
...
+    Error actual_size:Integer|Nothing limit:Integer
 
     ## PRIVATE
        Create a human-readable version of the error.
     to_display_text : Text
     to_display_text self =
         suggestion = " Use `Data.fetch` with `cache_policy=No_Cache`, or use `Data.download` to fetch the data to a local file, and `Data.read` to read the file."
-        "Response too large: repsonse size is over the limit ("+self.limit.to_text+")" + suggestion
+        actual_size_message = if self.actual_size == Nothing then "" else " "+self.actual_size.to_text
+        "Response too large: repsonse size"+actual_size_message+" is over the limit "+self.limit.to_text+"."+suggestion
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso]:
```suggestion
        "Response too large: response size"+actual_size_message+" is over the limit "+self.limit.to_text+"."+suggestion
```

File: std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java [hunk 2]
```diff
@@ -186,14 +189,14 @@ private File downloadResponseData(String cacheKey, Item item)
     boolean successful = false;
     try {
       // Limit the download to getMaxFileSize().
-      long maxFileSize = settings.getMaxFileSize();
-      boolean sizeOK = Stream_Utils.limitedCopy(inputStream, outputStream, maxFileSize);
+      long maxAllowedSize = Long.min(settings.getMaxFileSize(), getMaxTotalCacheSize());
```
Comment (radeusgd) [inline std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java]: very minor nitpick but I see the same expression `maxAllowedSize = Long.min(settings.getMaxFileSize(), getMaxTotalCacheSize());` twice here. Shall we just make it a `private` method and avoid the code duplication?

File: std-bits/base/src/main/java/org/enso/base/cache/ResponseTooLargeException.java [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
 
 public class ResponseTooLargeException extends Exception {
+  private final Long actualSize;
   private final long limit;
 
-  public ResponseTooLargeException(long limit) {
-    super("Response too large: repsonse size is over the limit (" + limit + ")");
+  public ResponseTooLargeException(Long actualSize, long limit) {
+    super(
+        "Response too large: repsonse size"
```
Comment (radeusgd) [inline std-bits/base/src/main/java/org/enso/base/cache/ResponseTooLargeException.java]:
```suggestion
        "Response too large: response size"
```

File: test/Table_Tests/src/IO/Fetch_Spec.enso [hunk 4]
```diff
@@ -457,6 +457,30 @@ add_specs suite_builder =
                     HTTP.fetch headers=headers2 uri
                     get_num_response_cache_entries . should_equal 2
 
+        group_builder.specify "Does not attempt to make room for the maximum file size when that is larger than the total disk size" pending=pending_has_url <| Test.with_retries <|
```
Comment (radeusgd) [inline test/Table_Tests/src/IO/Fetch_Spec.enso]: Test name seems misleading.
Where are we mocking disk size in this test? What I see is we are overriding **total cache size**, not disk space.

**Review by radeusgd — APPROVED** [review]
Looks good, just some suggestions

**Review by jdunkerley — APPROVED** [review]

File: std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java (hunk 2 above)
Comment (GregoryTravis) [reply std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java]: Done

File: test/Table_Tests/src/IO/Fetch_Spec.enso (hunk 4 above)
Comment (GregoryTravis) [reply test/Table_Tests/src/IO/Fetch_Spec.enso]: Yes -- changed to "total cache size".

**Pull request merged**